EIPS/eip-overlay_tree.md

```diff
@@ -0,0 +1,108 @@
+  * The fourth and final phase begins when the overlay trie is empty. The field holding its root is removed from the block header.
+
+## Motivation
+
+There is a long running interest in switching the state trie from a hexary format to a binary format, for reasons pertaining to proof and storage sizes. The conversion process poses a catch-up issue, caused by the sheer size of the full state: it can not be translated in a reasonable time (i.e. on the same order of magnitude as the block time). 
+
+## Specification
+The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).
+
+### Binary tries
+
+This EIP assumes that a binary trie is defined like the MPT, except that:
+
+  * The series of bytes in I₀ is seen as a series of _bits_ and so ∀i≤256, I₀[i] is the ith bit in key I₀
+  * The first item of an **extension** or a **leaf** is replacing nibbles with bits;
+  * A **branch** is a 2 item structure in which both items correspond to each of the two possible bit values for the keys at this point in their traversal;
+  * c(𝕴,i) ≡ RLP((u(0), u(1)) at a branch, where u(j) = n({I : I ∈ 𝕴 ⋀ I₀[i] = j}, i+1)
+
+### Phase 1
+
+Let _H₁_ be the previously agreed-upon block height at which phase 1 starts. For each block of height H₁ ≤ _h_ < H₂:
+
+  * Block headers contain a block header has a new Hₒ which is the _root of the overlay binary trie_
+  * Hᵣ ≡ P(H)ᵣ, i.e. the hexary trie root is the same as that of the block parent's. The hexary trie is referred to as the _hexary base trie_.
+
+The following is changed in the execution environment:
+
+  * Upon executing a _state read_, ϒ first searches for the address in the overlay trie. If the key can not be found there, ϒ then searches the hexary base trie as it did at block heights h' < H₁;
+  * Upon executing a _state write_, ϒ will insert or update the value into the overlay tree.
+
+A conversion process occurs in the background to turn the hexary trie into its binary equivalent. The end goal of this process is the calculation of the _root hash of the converted binary base trie_, denoted Hᵣ². The root of the hexary base trie is hereafter called Hᵣⁱ⁶.
+
+When a miner has calculated Hᵣ², it proceeds to phase 2.
+
+### Phase 2
+
+Phase 2 is the same as phase 1, except for the following:
+
+  * Hᵣ' ≡ Hᵣ²
+
+
+Phase 2 ends when a sufficient number of subsequent blocks have reported the same value for Hᵣ².
+
+### Phase 3
+
+The following changes occur in phase 3:
+
+  * N accounts are being deleted from the binary overlay trie and inserted into the binary base trie.
```

Comment:
Will `N` be specified later? Or would it happen implicitly via the next bullet-item?

Reply:
`N` still has to be specified. The way I see it, there are two approaches, and I'd like to get some feedback on that:

Approach 1: `N` is set to roughly 700. This is because a quick estimate based on my tests say it represents about 1 second. This number would be bigger if we use extensions in binary trees, which I will try when I complete the code that does this.

Approach 2: Dynamically calculate `N` based on the difficulty, to ensure it takes around 1 second.

Any thoughts?